actinia/mapset.py

```diff
@@ -23,31 +23,219 @@
         self.region = None
+        self.__location_name = location_name
         self.__actinia = actinia
         self.__auth = auth
         self.raster_layers = None
         self.vector_layers = None
         self.strds = None
 
+    def __request_url(
+        actinia_url,
+        location_name,
+        mapset_name=None,
+        task=None
+    ):
+        """
+        Provides the url to an Actinia mapset resource.
+
+        Parameters
+        ----------
+        actinia_url : str
+            The base url to actinia server
+        location_name : str
+            The GRASS location name
+            route: /locations/{location_name}/mapsets
+        mapset_name : str, default=None
+            The mapset name
+            route: /locations/{location_name}/mapsets/{mapset_name}
+        task : Enum(MAPSET_TASK), default=None
+            The requested task
+            (info) route:
+                /locations/{location_name}/mapsets/{mapset_name}/info
+            (lock) route:
+                /locations/{location_name}/mapsets/{mapset_name}/raster_layers
+            (raster_layers) route:
+                /locations/{location_name}/mapsets/{mapset_name}/raster_layers
+            (vector_layers) route:
+                /locations/{location_name}/mapsets/{mapset_name}/vector_layers
+            (strds) route:
+                /locations/{location_name}/mapsets/{mapset_name}/strds - GET
+            (processing) route:
+                /locations/{location_name}/mapsets/{mapset_name}/processing
+            (processing_async) route:
+                /locations/{location_name}/mapsets/{mapset_name}/processing_async
+
+        raster_layers : bool
+        Returns
+        -------
+        base_url : str
+            Return the url scheme for the mapset request
+        """
+        base_url = f"{actinia_url}/locations/{location_name}/mapsets"
+        if mapset_name is not None:
+            base_url = f"{base_url}/{mapset_name}"
+            if isinstance(task, MAPSET_TASK):
+                base_url = f"{base_url}/{mapset_name}{task.value}"
+
+        return base_url
+
+    def info(self):
+        """Get mapset info"""
+        if self.projection is None or self.region is None:
+            proc_res = self.request_info(
+                self.name,
+                self.__location_name,
+                self.__actinia,
+                self.__auth
+            )
+            self.projection = proc_res["projection"]
+            self.region = Region(**proc_res["region"])
+        return {"projection": self.projection, "region": self.region}
+
+    def delete(self):
+        """Deletes the mapset"""
+        self.delete_mapset_request(
+            self.name,
+            self.__location_name,
+            self.__actinia,
+            self.__auth
+        )
+        del self.__actinia.locations[self.__location_name].mapsets[self.name]
+
+    @classmethod
+    def list_mapsets_request(cls, location_name, actinia, auth):
```

Comment:
Should this method be used more often later? At the moment I don't see the point of moving the method. Was there a reason for this other than that it is now more sorted thematically? I would have made the class rather as a private class, because I do not know why the user should still use the method here otherwise from mapset.py.

Reply:
@anikaweinmann are you referring to the list_mapsets_request method? If so, my thought process for making the **classmethods** was so that a developer using the client wouldn't necessarily need to create a class instance if they do not need to, in addition to organizing all of the logic thematically by class.

Reply:
Okay. I understand and find the idea good.